packages/core/src/application.ts

```diff
@@ -312,6 +314,23 @@ export class Application extends Context implements LifeCycleObserver {
     this.setState('started');
   }
 
+  /**
+   * Register a function to be called when the application starts.
+   *
+   * This is a shortcut for adding a binding for a LifeCycleObserver
+   * implementing a `start()` method.
+   *
+   * @param fn The function to invoke, it can be synchronous (returning `void`)
+   * or asynchronous (returning `Promise<void>`).
+   * @returns The LifeCycleObserver binding created.
+   */
+  public onStart(fn: () => ValueOrPromise<void>): Binding<LifeCycleObserver> {
+    const key = `life-cycle-observers.${fn.name || '<onStart>'}-${uuid()}`;
+    return this.bind<LifeCycleObserver>(key)
+      .to({start: fn})
+      .tag(CoreTags.LIFE_CYCLE_OBSERVER);
```

Comment:
Let's use `.apply(asLifeCycleObserver)` for consistency. See https://github.com/strongloop/loopback-next/blob/master/packages/core/src/lifecycle.ts#L58

Reply:
Of course! Thanks for pointing out this binding template function to me :+1: